Source/WebCore/page/Page.cpp

```diff
@@ -2064,6 +2086,13 @@ void Page::setIsVisuallyIdleInternal(bool isVisuallyIdle)
     if (isVisuallyIdle == m_throttlingReasons.contains(ThrottlingReason::VisuallyIdle))
         return;
 
+#if PLATFORM(COCOA)
+    if (isVisuallyIdle && m_intervalStarted) {
+        workgroupRendering().finish();
+        m_intervalStarted = false;
+    }
+#endif
+
```

Comment:
Because the workgroup interval is scoped to Page::updateRendering, it's impossible in a well-formed program for a visually idle notification to be processed while we have a live interval -- since notifications are only processed when we return to the run loop. So, I think you can remove this.